scripts/generator/graphql_generator/csharp/Cart.cs.erb

```diff
@@ -26,9 +26,27 @@ namespace <%= namespace %> {
             }
         }
 
-        private CartLineItems _LineItems;
+        private bool IsSaved {
+            get {
+                return IsCreated && 
+                    LineItems.New().Count == 0 &&
+                    LineItems.Modified().Count == 0 &&
+                    LineItems.Deleted().Count == 0;
+            }
+        }
+
+        private bool IsCreated {
+            get {
+                return CheckoutId != null;
+            }
+        }
 
+        private const float POLL_DELAY = 0.5f;
+        private CartLineItems _LineItems;
         private ShopifyClient Client;
+        private string CheckoutId = null;
```

Comment:
Would you prefer that there was an abstraction for checkouts that you could pass a cart to? Yes right now we're merging the two concepts mainly to maintain the existing API which may not be a good reason to do this.